### PR TITLE
Fix defect #698 Default permissions on commands.log are 644, should change to 600

### PR DESCRIPTION
**Pull request by hu-weihua**
Depending on defect #698,  I changed the permissions of /var/log/xcat/commands.log to 600

### CODE REVIEW

**Comment by hu-weihua** [comment]
@neo954, I have changed code depending on your comments. Could you review again? thanks.

File: xCAT-server/sbin/xcatd [hunk 1]
```diff
@@ -1129,23 +1130,26 @@ unless ($cmdlog_svrpid){
...
+    print $cmdlogpidfile $$;
+    close($cmdlogpidfile);
+    xCAT::MsgUtils->trace(0,"I","xcatd: command log process $$ start");
+
     my $cmdlog_logfile_path=dirname($cmdlog_logfile);
     mkpath("$cmdlog_logfile_path") unless(-d "$cmdlog_logfile_path");
 
+    $cmdlog_logfile_umask = umask(0077);
     unless (open ($cmdlogfile, ">>$cmdlog_logfile")) {
+        umask($cmdlog_logfile_umask);
```
Comment (neo954) [inline xCAT-server/sbin/xcatd]: So, line 1143 is useless and can be removed. Right?

File: xCAT-server/sbin/xcatd [hunk 2]
```diff
@@ -1111,7 +1112,7 @@ unless ($cmdlog_svrpid){
         xCAT::MsgUtils->message("S","INFO xcatd: 'Command log writer' process $$ is trying to get port $cmdlog_port");
         my $pid = <$cmdlogpidfile>;
         if ($pid) {
-            kill 12, $pid;
+            kill 'USR2',  $pid;
```
Comment (neo954) [inline xCAT-server/sbin/xcatd]: One extra space appeared after the character comma, ",".

File: xCAT-server/sbin/xcatd [hunk 3]
```diff
@@ -1129,23 +1130,26 @@ unless ($cmdlog_svrpid){
...
+    xCAT::MsgUtils->trace(0,"I","xcatd: command log process $$ start");
+
     my $cmdlog_logfile_path=dirname($cmdlog_logfile);
     mkpath("$cmdlog_logfile_path") unless(-d "$cmdlog_logfile_path");
 
+    $cmdlog_logfile_umask = umask(0077);
     unless (open ($cmdlogfile, ">>$cmdlog_logfile")) {
+        umask($cmdlog_logfile_umask);
         xCAT::MsgUtils->trace(0,"E","xcatd: Can't open xcat command log file $cmdlog_logfile,command log process $$ stop.");
         if($cmdlogsvrlistener){close($cmdlogsvrlistener);}
```
Comment (neo954) [inline xCAT-server/sbin/xcatd]: Anyway, you are going to die. So this line is useless as well.

File: xCAT-server/sbin/xcatd [hunk 4]
```diff
@@ -1129,23 +1130,26 @@ unless ($cmdlog_svrpid){
...
         exit(0);
```
Comment (neo954) [inline xCAT-server/sbin/xcatd]: You ``die()`` here, due to the failure of ``open()``ing the log file. Thus, why ``exit(0)`` here. A non-zero exit code should be used.

**Comment by neo954** [comment]
Yep. Cool patch. Will merge this one.

**Pull request merged**